src/ddsrt/src/process/posix/process.c

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright(c) 2006 to 2018 ADLINK Technology Limited and others
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License
+ * v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+#include <stdio.h>
+#include <errno.h>
+#include <string.h>
+#include <unistd.h>
+#include <assert.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include "dds/ddsrt/process.h"
+
+
+ddsrt_pid_t
+ddsrt_getpid(void)
+{
+  /* Mapped to taskIdSelf() in VxWorks kernel mode. */
+  return getpid();
+}
+
+
+#ifdef DDSRT_USE_PROCESSCREATION
+
+#ifndef PIKEOS_POSIX
+#include <sys/wait.h>
+#endif
+
+#if defined(__APPLE__)
+#include <crt_externs.h>
+#else
+/* environ is a variable declared in unistd.h, and it keeps track
+ * of the environment variables during this running process. */
+extern char **environ;
+#endif
+
+#include "dds/ddsrt/heap.h"
+#include "dds/ddsrt/string.h"
+
+
+
+/*
+ * This'll take a argv and prefixes it with the given prefix.
+ * If argv is NULL, the new argv array will only contain the prefix and a NULL.
+ * The result array is always terminated with NULL.
+ */
+static char**
+prefix_argv(const char *prefix, char *const argv_in[])
+{
+  char **argv_out;
+
+  assert(prefix);
+
+  if (argv_in == NULL) {
+    argv_out = ddsrt_malloc(2 * sizeof(char*));
```

Comment:
@k0ekk0ek This statement is contradicted with the 
```
  if (ptr == NULL) {
    /* Heap exhausted */
    abort();
  }
```
check in all the allocation functions.
Or do you mean that these checks will be removed in the future and it would be nice if calling code is already prepared for that?

Reply:
Allocation results are now checked, despite the abort in the allocation functions.